src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingCook.java

```diff
@@ -141,6 +138,50 @@ public boolean canBeGathered()
         return builder.build();
     }
 
+    @Override
+    @Nullable
+    public IRecipeStorage getFirstRecipe(final Predicate<ItemStack> stackPredicate)
+    {
+        //First, do the normal check against taught recipes, and return those if found
+        IRecipeStorage storage = super.getFirstRecipe(stackPredicate);
+        if(storage != null)
+        {
+            return storage;
+        }
+
+        //If we didn't have a stored recipe, see if there is a smelting recipe that is also a food output, and use it. 
```

Comment:
I would only query this if ISFOOD

Reply:
Given that we don't know what the item is until after the query, I'm not sure how to limit it. We could do a combined predicate, though I don't see any advantage to that, and this code is clearer.

Reply:
I think there was one getFirstRecipe that gets an itemStack too. Where we could intercept.

Reply:
I had done that, and pulled it, because it's used rarely, and at the abstract level it just wraps the predicate version.  That is actually what I did first, and it didn't actually let things work, because the predicate is used almost everywhere.